ql/qldefines.hpp

```diff
@@ -215,6 +215,12 @@
 #endif
 // clang-format on
 
+/* suppress C++ code analysis warning C26812 in VS 2019:
+ * Prefer 'enum class' over 'enum' (Enum.3). */
+#if defined(BOOST_MSVC)
+    __pragma(warning(push))            \
+    __pragma(warning(disable : 26812))
+#endif
```

Comment:
This can go into `ql/config.msvc.hpp`, which contains the specific configs for VC++ and already disables some warnings.

Reply:
Also, you won't need the `#if defined` in that file.  You don't need the `push` either, since there's no `pop` afterwards.

Reply:
You're right. I have updated the PR accordingly.